src/plugins/typescript.js

```diff
@@ -1147,16 +1147,16 @@ export default (superClass: Class<Parser>): Class<Parser> => class extends super
     return super.parseSubscript(base, startPos, startLoc, noCalls, state);
   }
 
-  parseNewTypeArguments(node: N.NewExpression): void {
-    if (!this.isRelational("<")) { // tsTryParseAndCatch is expensive, so avoid if not necessary.
-      return;
+  parseNewArguments(node: N.NewExpression): void {
+    if (this.isRelational("<")) { // tsTryParseAndCatch is expensive, so avoid if not necessary.
+      // 99% certain this is `new C<T>();`. But may be `new C < T;`, which is also legal.
+      const typeParameters = this.tsTryParseAndCatch(this.tsParseTypeArguments.bind(this));
+      if (typeParameters) {
+        node.typeParameters = typeParameters;
```

Comment:
I think you should check that `this.match(tt.parenL)`, otherwise `new Foo<T>` will be allowed